File: src/main/java/com/larryhsiao/clotho/color/GreenInt.java
package com.larryhsiao.clotho.color;

import com.larryhsiao.clotho.Source;

/**
 * Extract blue int in color int
 */
public class GreenInt implements Source<Long> {
    private final long color;

    public GreenInt(long color) {this.color = color;}

    @Override
    public Long value() {
        return (color >> 8) & 0xff;
    }
}

File: src/main/java/com/larryhsiao/clotho/color/GreenInt.kt
package com.larryhsiao.clotho.color

import com.larryhsiao.clotho.Source

/**
 * Extract red int in color int
 */
class GreenInt(private val color: Long) : Source<Long> {
    override fun value(): Long {
        return color.shr(8) and 0xff
    }
}